File: api/models/document.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime
import uuid


class DocumentBase(BaseModel):
    filename: str
    size: int
    pages: int


class DocumentCreate(DocumentBase):
    pass


class Document(DocumentBase):
    id: str
    upload_date: datetime
    chunks_count: int
    status: str  # "processing", "ready", "error"

    class Config:
        from_attributes = True


class DocumentList(BaseModel):
    documents: List[Document]


class DocumentDelete(BaseModel):
    success: bool
    message: str


class UploadResponse(BaseModel):
    success: bool
    document_id: Optional[str] = None
    message: str
    status: str

File: frontend/src/types/document.ts
export interface Document {
  id: string;
  filename: string;
  size: number;
  pages: number;
  upload_date: string;
  chunks_count: number;
  status: string;
}

export interface DocumentList {
  documents: Document[];
}

export interface UploadResponse {
  success: boolean;
  document_id?: string;
  message: string;
  status: string;
}

export interface DocumentDelete {
  success: boolean;
  message: string;
}

export interface RagChatRequest {
  user_message: string;
  model?: string;
  api_key: string;
  document_id: string;
  k_chunks?: number;
}

export interface DocumentChunk {
  content: string;
  page: number;
  chunk_index: number;
  similarity_score: number;
}